packages/wrangler/src/r2/index.ts

```diff
@@ -50,10 +50,12 @@ async function createFileReadableStream(filePath: string) {
 	});
 }
 
-export function r2(r2Yargs: CommonYargsArgv) {
+export function r2(r2Yargs: CommonYargsArgv, subHelp: SubHelp) {
 	return r2Yargs
+		.command(subHelp)
 		.command("object", "Manage R2 objects", (r2ObjectYargs) => {
 			return r2ObjectYargs
+				.demandCommand()
```

Comment:
just to make sure this is what you want here...

calling `demandCommand()` will throw an error if `wrangler r2 object` is called without any options. So basically this 👇 
<img width="789" alt="Screenshot 2024-06-21 at 17 59 19" src="https://github.com/cloudflare/workers-sdk/assets/4638332/d7f2afc2-3ccc-4674-9be4-a39107620754">

An alternative would be to instead call `.command(subHelp)` which would ensure we still show the help menu, without actually throwing. So basically this 👇 
<img width="758" alt="Screenshot 2024-06-21 at 18 00 46" src="https://github.com/cloudflare/workers-sdk/assets/4638332/9671ab99-563f-4647-9663-fa6025f6c382">

As far as I can see this type of behaviour is not standardized across the wrangler codebase, with commands such as `ai` and `d1` using `.demandCommand`, and all the other not. With the lack of a clear standard UX, I'm not sure what to advise here. I guess either of the two options are OK, but perhaps worth discussing with the r2 team/product what you'd like this behaviour to be?

Reply:
Just confirmed with the team, and it's fine as is.